index.bs

```diff
@@ -5922,6 +5932,144 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+        ;
+        ; Note that there are details dependent upon the particular attestation
+        ; statement format.
+        ; See https://www.w3.org/TR/webauthn/#sctn-defined-attestation-formats.
+
+        $$attStmtType,
+    }
+
+    ```
+
+: Authenticator extension processing
+:: For both [=authenticatorMakeCredential=] and [=authenticatorGetAssertion=] operations:
+    1. Create or select the [=user credential=] as usual, and let `userCredentialId` be its <code>[=credentialId=]</code>.
+
+    1. If a [=hardware-bound device key pair=] does not already exist for this [=user credential=] on the [=authenticator=], create it using the same public key algorithm as that used by the [=user credential=]'s [=credential key pair=], otherwise locate the existing [=device-bound key=].
+
+    1. Let `dpk` be the newly created or existing [=device public key=], in COSE_Key format [=credentialPublicKey|in the same fashion as for the user credential's credentialPublicKey=] when the latter is conveyed in [=attested credential data=].
+
+    1. Let `devicePrivateKey` be the newly created or existing [=device private key=].
+
+    1. Let `clientDataHash` be the [=hash of the serialized client data=].
+
+    1. Let `sig` be the result of signing the concatenation of `clientDataHash` and `userCredentialId` using the `devicePrivateKey` and the signature algorithm appropriate for the `devicePrivateKey`'s public key algorithm: `sign((clientDataHash || userCredentialId), devicePrivateKey)`.
+
+    1. Let `scope` have the value zero (0x00) if this is an "entrie-device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
+
+    1. Let `$$attStmtType` be the result of generating an attestation statement in the [=attestation statement format=] appropriate for the [=user credential=], although substituting the authenticator's <code>[=aaguid=]</code> for |authenticatorData|, and substituting `userCredentialId` for |clientDataHash| in the [=attestation statement format=]'s signing procedure (see [[#sctn-generating-an-attestation-object]]). Attestation statement formats define the `fmt` and `attStmt` members of `$$attStmtType`'s value.
+
+        In summary, `$$attStmtType`'s value typically contains a signature value calculated over the bytes of `(clientDataHash || userCredentialId)`, the attestation certificate or public key, and supporting certificates, if any.
+
+        Note: The details of `$$attStmtType`'s value are dependent upon the particular [=attestation statement=] format. See [[#sctn-attestation-formats]].
+
+    1. Let `AttObjForDevicePublicKey` be a [=CBOR=] object, as defined above (in "Authenticator extension output"), whose values for its `sig`, `aaguid`, `dpk`, `scope`, and `$$attStmtType` members are as calculated in the prior steps.
+
+    1. Let  `AttObjForDevicePublicKey` be the value for the `devicePubKey` member of `$$extensionOutput`, and return the extension output.
```

Comment:
This also seems to conflate the `attObjForDevicePublicKey` type and its value. Maybe something like this is better?

```suggestion
    1. Let the `devicePubKey` [=authenticator extension output=] be a [=CBOR=] map as defined by `attObjForDevicePublicKey` above, whose values for its `sig`, `aaguid`, `dpk`, `scope`, and `$$attStmtType` members are as calculated in the prior steps.
```

Reply:
another good catch!